File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int waterFill = Integer.MIN_VALUE;
        int si = 0, ei = height.length-1;
        int areaTemp = 0;
        while (si < ei){
            if(height[si] < height[ei]){
                areaTemp = height[si] * (ei-si);
                si++;
            }else{
                areaTemp = height[ei] * (ei - si);
                ei--;
            }
            waterFill = Math.max(waterFill,areaTemp);
        }
        return waterFill;
    }
}

File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        si = 0
        ei = len(height)-1
        areaTemp = 0
        maxArea = 0
        while si < ei:
            if height[si] < height[ei]:
                areaTemp = height[si] * (ei-si)
                si+=1
            else:
                areaTemp = height[ei] * (ei-si)
                ei-=1
            if areaTemp >= maxArea:
                maxArea = areaTemp
        return maxArea